Added MySQL Database Support (updated pyproject dependencies)

CodeStrate opened the pull request:
- Added mysql connector dependency and updated README and pyproject to reflect it (similar to postgresql)
- Added utility to parse individual connection params from DSN for MySQL
- Migration Logic borrowed from SQLite with MySQL syntax specific changes

CodeStrate committed:
Added MySQL Database Support (updated pyproject dependencies)

s-bose commented on wandern/databases/mysql.py:
```diff
@@ -0,0 +1,258 @@
+from urllib.parse import urlparse, parse_qs
+from typing import Dict, Any
+
+def parse_params_from_dsn(dsn: str) -> Dict[str, Any]: # str key, Any value
```
Please use normal `dict[str, Any]` with generic parameters instead of `typing.Dict`

CodeStrate replied:
Added a TypedDict for Connection Params instead of Dict.

s-bose commented on wandern/databases/mysql.py:
```diff
@@ -0,0 +1,258 @@
+    except SyntaxError:
+        raise
+    except Exception as e:
+        raise Exception(f'Encountered an issue: {e}')
```
Re-raising generic exceptions is not good. We are already handling generic exception during `connect()`

s-bose reviewed (changes requested):
- Documentations missing.
- Several issues with code structure and refactor
- Unit tests missing.

Overall nice work! Please fix the highlighted bits and add comprehensive unit tests.

CodeStrate commented:
Could you elaborate on the documentation part?

CodeStrate committed:
Updated Docs to mention MySQL specific DSN workaround.

CodeStrate committed:
Changes:
- Removed unnecessary comments.
- Added a MySQLConnectionParams typedDict in place of typing.Dict
- Reworked generic exceptions to be more specific. and removed unnecessary exceptions.
- Removed defaults in params and structured query string params better.

CodeStrate committed:
Added Unit tests for MySQL using SQLite and Postgres as a base.

CodeStrate committed:
Changes:
- Changed TEXT to VARCHAR with fixed length as per MySQL syntax.
- Fixed a typo in providers to implement MySQLProvider correctly.
- Fixed Unread result errors while testing by correctly consuming cursor results.

CodeStrate commented:
I need your assistance since unit testing is not my strong suit, any suggestions on the autocommit test might help.

CodeStrate committed:
Removed bare try-except blocks, changed timestamp according to MySQL syntax and added autocommit True to match Postgres.

CodeStrate reviewed (commented):
All requested changes are made @s-bose . Please review

CodeStrate committed:
Removed mysql libraries from direct dependencies.
- Also removed self.connection_params making it a local variable.

CodeStrate commented on wandern/databases/mysql.py:
```diff
@@ -0,0 +1,324 @@
+        """
+        try:
+            params = parse_params_from_dsn(self.config.dsn)
+            self.connection_params = validate_parsed_params(params)
```
made it a local variable and tested. no issues there.

CodeStrate commented on pyproject.toml:
```diff
@@ -12,10 +12,12 @@ classifiers = [
 ]
 dependencies = [
     "jinja2>=3.1.6",
+    "mysql-connector-python>=9.4.0",
```
removed direct dependencies for mysql and testcontainers[mysql]. optional dependency is retained.

CodeStrate commented:
@s-bose Commit 12acbc6 fixes the two nitpicks you had. Please review and LMK if looks good to merge.

CodeStrate committed:
Added uv lock file.

s-bose commented:
@CodeStrate There is a failing test.

https://github.com/s-bose/wandern/actions/runs/18948964620/job/54108342466

CodeStrate committed:
Added mysql test in test_provider and changed mysql to mssql in unsupported testcase to avoid failure.

s-bose commented:
@CodeStrate Two more things.

```
wandern/databases/mysql.py#L29-L90

Complex Method
```

Could you please try to shorten the code a bit?
And, could you please bump the package version to the next minor release?

Sorry for being a pain! :)

CodeStrate committed:
Changes:
- Bumped Package version to 0.0.4
- Simplified the parser function in mysql.py and modified the tests accordingly.

CodeStrate commented:
@s-bose with 095431a CodeFactor doesn't report an error now. I simplified the parser code, changed the 2 tests for missing host and port (combined them into one since we checked for either in the parser as both are required, and we don't assume any default fallbacks)

also bumped 0.0.3 to 0.0.4 :)
